File: WEEK1/003/ecole003.js
const n = parseInt(require('fs').readFileSync('/dev/stdin').toString().trim());

let count = 1;
let start = 1, end = 1;
let sum = 1;

while (end !== n) {
    if (sum === n) {
        count++;
        end++;
        sum += end;
    } else if (sum < n) {
        end++;
        sum += end;
    } else { // sum > n
        sum -= start;
        start++;
    }
}
console.log(count);

File: WEEK1/003/ecole003.py
n = int(input())
count = 1
start, end = 1, 1
current_sum = 1

while end != n:
    if current_sum == n:
        count += 1
        end += 1
        current_sum += end
    elif current_sum < n:
        end += 1
        current_sum += end
    else:  # current_sum > n
        current_sum -= start
        start += 1

print(count)
